### PR TITLE
Fix incorrect styling when <select> element is present

### LINKED ISSUE #4141: Font Style doesn't get applied correctly to styled header used together with form element
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Setup CKEditor with below configuration:
```
var editor = CKEDITOR.replace( 'editor1', {
	extraAllowedContent : 'h4{*}'
});
```
2. Switch to source mode and paste the following HTML
```
<div>
	<select>
	  <option>Other Languages</option>
	</select>
</div>
<div>
	<h4 style="font-family:arial,sans-serif;">Key Points</h4>
</div>
```
3. Press `Ctrl+A` to select all content and change the font name to Comic Sans MS

### Expected result

Text style in header gets changed to Comic Sans MS

### Actual result

Text style in header doesn't get changed to Comic Sans MS.

To be more precise, with this particular HTML, whole `div` gets wrapped in `span` and you will only get `select` and `h4` elements wrapped as expected if you switch to source mode and back or you get data from the editor using `getData` method.
obraz
obraz

Problem can be reproduced from CKEditor 4.0.0.

This doesn't happen when `h4` has no `font-family` styling or when select tag is not present.

## Other details

* Browser: Firefox, IE11
* OS: Any
* CKEditor version: 4.0.0+
* Installed CKEditor plugins: Font, Forms

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#4141](https://github.com/ckeditor/ckeditor4/issues/4141): Fixed: styles are incorrectly applied when there is `<select>` element inside the editor.
```

## What changes did you make?

I've added skipping for `select` and `option` elements – this way these elements are not put inside `span`s with styles.

The issue was connected with the fact that our style engine iterated through `select` and its `option`s, trying to add styling to them. This resulted in incorrect selection, jumping outside the paragraph with `select`. Skipping `select` & `option` altogether eliminates the issue.

## Which issues does your PR resolve?

Closes #4141.

### CODE REVIEW

**Review by jacekbogdanski — CHANGES_REQUESTED** [review]
Looks to work correctly.

I'm wondering if we could make this change less coupled with the `style.js` module. `style.js` already exposes `data-nostyle` attribute which probably could be used on a form elements. Could you check if we are able to use that attribute instead?

If it's not possible, maybe we could go with some more orthogonal approach by exposing API for ignoring chosen elements?

File: tests/core/style/manual/selectfontstyles.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<body>
+<div id="editor">
+	<p>
+		<select>
+			<option>Some option</option>
+		</select>
+	</p>
+	<h4 style="font-size:40px;font-family:arial,sans-serif;">Heading</h4>
```
[the remark this answers is not in the document]
Comment (Comandeer) [reply tests/core/style/manual/selectfontstyles.html:8]: TBH I wouldn't touch it. Selection of form elements is broken in some way in de facto every browser and I'm for replacing them with fake objects (just like we do for `iframe`s and other "weird" things).

**Comment by Comandeer** [comment]
I went with simpler approach and move the array of unstylable elements to `CKEDITOR.style` property – thanks to that it's configurable now.

File: tests/core/style/manual/selectfontstyles.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (Comandeer) [reply tests/core/style/manual/selectfontstyles.html:8]: I created an issue for it – #4211.

File: tests/core/style/applyremove.js [hunk 2]
```diff
@@ -1053,8 +1075,14 @@
 
 	bender.test( tcs );
 
-	function createInlineStyleTestCase( fixtureId ) {
+	function createInlineStyleTestCase( fixtureId, options ) {
+		options = options || {};
+
 		return function() {
+			if ( options.ignore && options.ignore ) {
```
Comment (hub33k) [inline tests/core/style/applyremove.js:1082]:
```suggestion
			if ( options.ignore ) {
```

**Comment by Comandeer** [comment]
I introduced some changes after F2F conversation with @jacekbogdanski and moved marking `select, option` as ignored to `forms` plugin.

**Review by jacekbogdanski — APPROVED** [review]
Looks good!

**Pull request merged**